### DISCUSSION TITLE
LogicFormula or other component turn a valid boolean expression to invalid one.

### DISCUSSION THREAD

**Discussion by xwaeaewcrhomesysplug** [axkr/symja_android_library#215] [Q&A] [answered]
**Describe the bug**
inner antlr converted logicformula instead failed?

**To Reproduce**
start the java evaluator,engine.
TautologyQ(a&b)

Example input Symja input expression:
TautologyQ(a&b)

Error result:

TautologyQ: illegal arguments: "b*(a&)" in LogicFormula
TautologyQ(b*(a&))

**Expected behavior**
just works and does not turn into,b*(a&)

Expected (correct) result expression:

```
TautologyQ(a&b)
false
```

**Java**
 - openjdk version "15" 2020-09-15
OpenJDK Runtime Environment (build 15+36-1562)
OpenJDK 64-Bit Server VM (build 15+36-1562, mixed mode, sharing)

 -linux,fedora,64bit

**Comment by axkr** [comment, marked as answer]
You have to use "doubled character" `&&` for `And()` and  `||` for `Or()`.
Only one `&` is used as an operator for pure functions.

See:
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/And.md
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/Or.md
- https://github.com/axkr/symja_android_library/blob/master/symja_android_library/doc/functions/Function.md
